### ISSUE TITLE
Empty area at the end of the widget

### ISSUE THREAD

**Issue by de-es** [andstatus/todoagenda#67]
There is always an empty area at the end of the widget after the upcoming events, even when there is none upcoming. When you tab on it, it takes you to settings screen of the widget.

Is this intended or a bug? If intended, is there a setting to hide it?

Example (yellow are the upcoming events for the next day):

![todo](https://user-images.githubusercontent.com/7645771/107585551-5d842400-6bf6-11eb-9377-ffd4a195764e.png)

Using version 4.4.0

**Comment by yvolk** [comment]
Yes, this empty list item was added intentionally to give a visual clue that there are no more events below it. I.e. it is easier to understand that you scrolled to the bottom of the list.
No options to remove it.
See also #36

**Comment by de-es** [comment]
Thanks for your feedback.

Personally I can live with that. But from an UX point of view, it's quite confusing. You touch an empty, colored space to get to the app settings. Imho the not so tech-savvy (or touch-safe) user wouldn't expect that.

Wouldn't just a simple (colored?) line be a prefered solution?

**Comment by yvolk** [comment]
> Personally I can live with that. But from an UX point of view, it's quite confusing. You touch an empty, colored space to get to the app settings. Imho the not so tech-savvy (or touch-safe) user wouldn't expect that.

There are TWO main ways to get to the widget's settings:
1. An icon on the widget's header
2. The "ToDo Agenda" app in the Android Launcher.

Nobody knows what to expect from clicking on a blank area of the widget. So why not Settings?

**Comment by kyledev80020** [comment]
I too considered this a "bug," and I still do if there's not a way to turn it off.  

A helpful? suggestion though, I decided to "disable" it by having the future color be 100% transparent.

**Comment by yvolk** [comment]
> I too considered this a "bug," and I still do if there's not a way to turn it off.

Any feature that cannot be turned off is a bug for you?!  I see such an opinion from time to time, regarding different apps...
Why not? But **any** app has lots of "bugs" that will never be fixed then :-)

**Comment by yvolk** [comment]
@de-es @kyledev80020 Recently I added "Max number of events" option https://github.com/andstatus/todoagenda/issues/139
and if that option is turned on (i.e. the maximum number is set) AND actual number of events is not less than that number
then the last blank entry is not added.

**Comment by de-es** [comment]
Thanks for coming back to this. Limiting the number of events wouldn't be my personal use case.

But what about linking the empty area to the calendar intent? It would imgo be more convenient than settings.

**Comment by yvolk** [comment]
Hi @de-es,
"Max number of events" option does exactly what you need: it removes the empty row at the end of the list.
Please try it!
You may set 10000 as a limit if you worry about filtering out any needed events.

>  linking the empty area to the calendar intent

Please clarify this.

**Comment by de-es** [comment]
Thanks @yvolk 

I did test the new feature.

My general use case for the widget is showing all today's events. If there are more events than the widget could show, scrolling is permitted.

I could limit the number of events now and won't get the empty area at the end, but limiting is not what I want for my use case. If I set the limit to an unlikely high number, the empty area is still present.

> linking the empty area to the calendar intent

Instead of opening the settings screen, it would be more convenient to open the calendar. Like when I tab on "No upcoming events".

**Comment by yvolk** [comment]
@de-es Thank you for testing.
I see why setting "Max number of events" to some large value doesn't turn off the "LastEntry": https://github.com/andstatus/todoagenda/commit/0f0eaea6fe75740739002766db014b96cf41b299#diff-74e89656f3ea6ffcf4089c1cfe23bda338d159810c112f4b6fa2f658c7469606R172
- It is turned off only in a case actual number of events equals to this "Max number of events" value :-(
Really not good enough for your case...
Looks like we need a separate option for that "Empty area"...

**Comment by yvolk** [comment]
@de-es Please check v.4.12.2 at #3 - now the Empty area should not be present for any positive value of "Max number of events"

**Comment by de-es** [comment]
@yvolk Works perfectly.  🥳 Thank you very much for implementing.
